File: Trabalho Prático JS/model/Cliente.js
import { ContaCorrente } from "../model/ContaCorrente.js";
import { ContaPoupanca } from "../model/ContaPoupanca.js";
import { escreverLinha } from "../view/InterfaceBanco.js";

export class Cliente {
    #cpf;
    #nome;
    #sobrenome;
    #contas;

    constructor(nome, sobrenome, cpf) {
        this.#nome = nome;
        this.#sobrenome = sobrenome;
        this.#cpf = cpf;
        this.#contas = new Array();
    }

    // Getter & Setters
    get nome(){
        return this.#nome;
    }
    set nome(nome) {
        this.#nome = nome;
    }

    get sobrenome(){
        return this.#sobrenome;
    }
    set sobrenome(sobrenome) {
        this.#sobrenome = sobrenome;
    }

    get contas(){
        return this.#contas
    }

    // Methods
    adicionarConta(conta) {
        this.#contas.push(conta);
    }
    getConta(indice) {
        return this.#contas[indice];
    }
    getNumeroContas() {
        return this.#contas.length;
    } 
    exibirCliente() {
        escreverLinha("\nCliente [" + this.#nome + ", " + this.sobrenome + "]" + " possui " + this.getNumeroContas() + " conta(s).");
        let tipoConta;
        let i = 0;
        this.contas.forEach(conta => {
            i++;
            if(conta instanceof ContaPoupanca) {
                tipoConta = "Conta Poupança";
            } else if(conta instanceof ContaCorrente) {
                tipoConta = "Conta Corrente";
            } else {
                throw new Error("Conta não reconhecida.");
            }
            escreverLinha("\tConta " + i  + ": " + tipoConta + " com saldo de " + conta.saldo.toLocaleString('pt-br', {style: 'currency', currency: 'BRL'}));
        })
    }
}

File: exercicio13/Banco3/src/main/java/banco/Cliente.java
package banco;

import java.util.ArrayList;

public class Cliente {
    private String nome;
    private String sobrenome;

    Conta conta;

    private ArrayList<Conta> contas = new ArrayList<>();

    public Cliente(String nome, String sobrenome) {
        this.nome = nome;
        this.sobrenome = sobrenome;
    }

    public String getNome() {
        return nome;
    }

    public void setNome(String nome) {
        this.nome = nome;
    }

    public String getSobrenome() {
        return sobrenome;
    }

    public void setSobrenome(String sobrenome) {
        this.sobrenome = sobrenome;
    }

    public Conta getConta() {
        return conta;
    }

    public void adicionarConta(Conta conta){
        this.contas.add(conta);
    }

    public Conta getConta(int id) {
        return contas.get(id);
    }

    public void setConta(Conta conta) {
        this.conta = conta;
    }

    public int getNumeroDeContas(){
        return contas.size();
    }
}
